### PR TITLE
Migrate Fake.Api.Slack to NET Core

### PR DESCRIPTION
**Pull request by vlukash**
# Migrate Fake.Api.Slack to NET Core

### Files:
`SlackNotificationHelper.fs `

### New namespace:
`Fake.Api`

### New module name:
`Slack`

### New *.fsx script example:
```
open Fake.Core
open Fake.Api

Target.Create "SendMsgToSlack" (fun _  ->
    let webhookUrl = "https://hooks.slack.com/services/some/random/text"
    Slack.SendNotification webhookUrl (fun p ->
        {p with
            Text = "My Slack Notification!\n<https://google.com|Click Here>!"
            Channel = "#general"
            Icon_Emoji = ":ghost:"
            Attachments = [| 
                {Slack.NotificationAttachmentDefaults with
                    Fallback = "Attachment Plain"
                    Text = "Attachment Rich"
                    Pretext = "Attachment Pretext"
                    Color = "danger"
                    Fields = [|
                        {Slack.NotificationAttachmentFieldDefaults with
                            Title = "Field Title 1"
                            Value = "Field Value 2"}
                        {Slack.NotificationAttachmentFieldDefaults with
                            Title = "Field Title 1"
                            Value = "Field Value 2"}|]
                }
                {Slack.NotificationAttachmentDefaults with
                    Fallback = "Attachment 2 Plain"
                    Text = "Attachment 2 Rich"
                    Pretext = "Attachment 2 Pretext"
                    Color = "#FFCCDD"
                }|]
        })
    |> printfn "Result: %s"
)

Target.RunOrDefault "SendMsgToSlack"
```

### Tested on:
- Ubuntu 16.04, mono 5.2.0.215
- Ubuntu 16.04, NET Core 2.0

### CODE REVIEW

**Comment by matthid** [comment]
Thanks for this it looks really nice!
Sorry for the delay. There were some problems. 

Currently master is "green" (there is still a transient error from time to time). Please try to rebase/merge to get this green.

About the namespace: `Fake.Api` is probably fine. I'm currently thinking about where to add `Octokit` (Github API). But `Fake.Api.Github` might just be the right place. I'm just a bit afraid that "Api" might be too general? What do you think?

**Comment by vlukash** [comment]
@matthid thanks for fixing the build!

About the `Fake.Api` namespace. Yes, probably it's a good idea to create more specific namespaces. What about a `Fake.Integrations` for the Slack?
Please let me know your options/preference for the namespace so I can update it if needed.

File: src/app/Fake.Api.Slack/SlackNotification.fs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+open System.Net.Http
+#else
+open System.Net
+#endif
+open Newtonsoft.Json
+
+/// Contains a task to send notification messages to a [Slack](https://slack.com/) webhook
+module Slack = 
+    /// The Slack notification attachment field parameter type
+    [<CLIMutable>]
```
Comment (matthid) [inline src/app/Fake.Api.Slack/SlackNotification.fs]: I kind of removed `[<CLIMutable>]` when porting records - it had historical reasons, but was not actually used.

File: src/app/Fake.Api.Slack/SlackNotification.fs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        /// (Required) The field title
+        Title: string
+        /// (Required) Text value of the field
+        Value: string
+        /// Whether the value is short enough to be displayed side-by-side with other values
+        Short: bool
+    }
+    
+    /// The Slack notification attachment parameter type
+    [<CLIMutable>]
```
Comment (matthid) [inline src/app/Fake.Api.Slack/SlackNotification.fs]: same here

File: src/app/Fake.Api.Slack/SlackNotification.fs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    type NotificationAttachmentParams = {
+        /// (Required) Text summary of the attachment that is shown by clients that understand attachments but choose not to show them
+        Fallback: string
+        /// The title of the attachment
+        Title: string
+        /// Content to which the title should link
+        Title_Link: string
```
Comment (matthid) [inline src/app/Fake.Api.Slack/SlackNotification.fs]: hm should we really use `_` here?

File: src/app/Fake.Api.Slack/SlackNotification.fs [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        /// Text that should appear above the formatted data
+        Pretext: string
+        /// Color of the attachment text. Can be hex-value(e.g. "#AABBCC") or one of "'good', 'warning', 'danger'.
+        Color: string
+        /// Text to be displayed as a table below the message
+        Fields: NotificationAttachmentFieldParams[]
+    }
+    
+    /// The Slack notification parameter type
+    [<CLIMutable>]
```
Comment (matthid) [inline src/app/Fake.Api.Slack/SlackNotification.fs]: remove

File: src/app/Fake.Api.Slack/SlackNotification.fs [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    let private SerializeData data =
+        JsonConvert.SerializeObject(data, Formatting.None, new JsonSerializerSettings(NullValueHandling = NullValueHandling.Ignore, ContractResolver = lowerCaseContractResolver))
+    
+    /// Sends a notification to a Slack Channel
+    /// ## Parameters
+    ///  - `webhookURL` - The Slack webhook URL
+    ///  - `setParams` - Function used to override the default notification parameters
+    let SendNotification (webhookURL : string) (setParams: NotificationParams -> NotificationParams) =
+        let sendNotification param =
+            #if NETSTANDARD
```
Comment (matthid) [inline src/app/Fake.Api.Slack/SlackNotification.fs]: please don't indent compiler directives

File: src/app/Fake.Api.Slack/SlackNotification.fs (hunk 1 above)
Comment (vlukash) [reply src/app/Fake.Api.Slack/SlackNotification.fs]: Got it, removed.

File: src/app/Fake.Api.Slack/SlackNotification.fs (hunk 5 above)
Comment (vlukash) [reply src/app/Fake.Api.Slack/SlackNotification.fs]: updated

File: src/app/Fake.Api.Slack/SlackNotification.fs (hunk 3 above)
Comment (vlukash) [reply src/app/Fake.Api.Slack/SlackNotification.fs]: updated

**Comment by vlukash** [comment]
@matthid thanks for the review.
Removed [<CLIMutable>] and `_`.

**Pull request merged**